engine/src/witness/common/cf_parameters.rs

```diff
@@ -44,4 +56,41 @@ mod tests {
 			MAX_VAULT_SWAP_PARAMETERS_LENGTH as usize >= VaultSwapParameters::max_encoded_len()
 		);
 	}
+
+	#[test]
+	fn test_versioned_cf_parameters() {
+		let vault_swap_parameters = VaultSwapParameters {
+			refund_params: ChannelRefundParameters {
+				retry_duration: 1,
+				refund_address: ForeignChainAddress::Eth(sp_core::H160::from([2; 20])),
+				min_price: Default::default(),
+			},
+			dca_params: None,
+			boost_fee: 0,
+			broker_fee: Beneficiary { account: AccountId::new([3; 32]), bps: 4 },
+			affiliate_fees: sp_core::bounded_vec![],
+		};
+
+		let cf_parameters = CfParameters::<()> {
+			ccm_additional_data: (),
+			vault_swap_parameters: vault_swap_parameters.clone(),
+		};
+
+		let mut encoded = VersionedCfParameters::V0(cf_parameters).encode();
+
+		assert_eq!(encoded, REFERENCE_EXPECTED_ENCODED);
+
+		let ccm_cf_parameters = CfParameters {
+			ccm_additional_data: CcmAdditionalData::default(),
+			vault_swap_parameters,
+		};
+
+		encoded = VersionedCcmCfParameters::V0(ccm_cf_parameters).encode();
+
+		// Extra byte for the empty ccm metadata
+		let mut expected_encoded = vec![0];
+		expected_encoded.extend_from_slice(REFERENCE_EXPECTED_ENCODED);
```

Comment:
Nit: I believe you can just convert to Vec with `Vec::from` (or alternatively compare slices)